app/src/main/java/org/mozilla/focus/observer/LoadingObserver.java

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.focus.observer;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.os.SystemClock;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import org.mozilla.focus.architecture.NonNullObserver;
+import org.mozilla.focus.session.Session;
+import org.mozilla.focus.telemetry.TelemetryWrapper;
+
+public class LoadingObserver {
+
+    private static final String LOG_TAG = "LoadingObserver";
+
+    public LoadingObserver(LifecycleOwner owner, final Session session) {
```

Comment:
\[PMD\] INFO: The constructor 'LoadingObserver' has a Cyclomatic Complexity of 5\.